instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/MainJarPathFinder.java

```diff
@@ -44,9 +44,13 @@ Path detectJarPath() {
   @Nullable
   private Path getJarPathFromProcessHandle() {
     String[] javaArgs = getProcessHandleArguments.get();
-    for (int i = 0; i < javaArgs.length; ++i) {
-      if ("-jar".equals(javaArgs[i]) && (i < javaArgs.length - 1)) {
-        return Paths.get(javaArgs[i + 1]);
+    boolean jarOptionFound = false;
+    for (String javaArg : javaArgs) {
+      if ("-jar".equals(javaArg)) {
+        jarOptionFound = true;
+      }
+      if (jarOptionFound && !javaArg.startsWith("-")) {
```

Comment:
perhaps use `else if`. There should also be a comment describing why this is done. Something along the line of `// flags can appear between -jar and the jar path, ignore them`

Reply:
thanks for your suggestion.